habitat_baselines/rl/ppo/ppo.py

```diff
@@ -4,20 +4,71 @@
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
 
+import contextlib
 from typing import Optional, Tuple
 
 import torch
 from torch import Tensor
 from torch import nn as nn
 from torch import optim as optim
 
+from habitat import logger
 from habitat.utils import profiling_wrapper
 from habitat_baselines.common.rollout_storage import RolloutStorage
+from habitat_baselines.rl.ddppo.algo.ddp_utils import rank0_only
 from habitat_baselines.rl.ppo.policy import Policy
+from habitat_baselines.utils.common import cast_to_float_if_half
 
 EPS_PPO = 1e-5
 
 
+class FP16OptimParamManager:
```

Comment:
Does it work for PPO only? If not let's move to common and give it a docstring how to use it.

Reply:
It's general-ish.  I made my own as I haven't found one we could rely on as a dependency instead.  fairscale may eventually come to fit that.  My hesitation for moving it to common is that things like this naturally have a lot of sharp edges

Reply:
Agree with @erikwijmans Apex is a pain.